core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -858,6 +864,11 @@ public static long roundReal(@SqlType(StandardTypes.REAL) long num, @SqlType(Sta
         if (rescaledRound != Long.MAX_VALUE) {
             result = sign * (rescaledRound / factor);
         }
+        else if (rescaled == Double.POSITIVE_INFINITY) {
+            double positive = sign * numInFloat;
+            BigDecimal bd = BigDecimal.valueOf(positive);
+            result = sign * bd.setScale((int) decimals, RoundingMode.HALF_UP).doubleValue();
```

Comment:
We probably need to use `Math.toIntExact` (and catch `ArithmeticException`, converting it to `TrinoException` if `decimals > Integer.MAX_VALUE`)